pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -217,7 +173,30 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 				continue
 			}
 			clusterKey := buildClusterKey(service, port, cb, proxy, efKeys)
-			cached, allFound := cb.getAllCachedSubsetClusters(*clusterKey)
+			// clusterKey can't be nil as per buildClusterKey
+			subsetKeys := cb.getSubsetKeys(*clusterKey)
+			// it exists, we shouldn't mark as deleted
+			deleted.Delete(clusterKey.clusterName)
+			// we can only skip generation of the clusters associated with this service iff none of the updated
+			// resources depend on that cluster
+			canSkip := skipForDelta(clusterKey, updated) && skipForDelta(oldCacheKeys[clusterKey.clusterName], updated)
+			// the same applies for all subsets
+			for _, ss := range subsetKeys {
+				skipSubset := skipForDelta(&ss, updated) && skipForDelta(oldCacheKeys[ss.clusterName], updated)
```

Comment:
So the thing that is broken (and fails on `TestTraffic/autopassthrough`) is that we now store a cache key. We then look at the cache key and see if it dependsOn the changed types. However, dependsOn is not the only way cache keys are invalidated - Clear/ClearAll can be called on the cache. When this happens, we would still have all of these cache keys in the `oldCacheKeys` map, and we would see the update from some type and get `skipForDelta`. But that is wrong -- we cannot skip it because the cache key is invalidated (we just didn't notice).

Reply:
added invalidation logic